src/replay/index.ts

```diff
@@ -420,6 +425,52 @@ export class Replayer {
     }
   }
 
+  private setUpAutoFit() {
+    var observer = new MutationObserver((mutations) => {
+      mutations.forEach((mutation) => {
+        if (mutation.type == "attributes") {
+          console.log("test attributes changed", mutation)
+          this.diggestAutoFit()
+        }
+      });
+    });
+    
+    observer.observe(this.wrapper, {
+      attributes: true
+    });
+
+    this.diggestAutoFit()
+
+    return observer
+  }
+
+  private diggestAutoFit() {
+    this.offsetMarginFit = null
+    this.offsetFactorFit = 1
+
+    if(!this.wrapper.hasAttribute('data-fit') || this.wrapper.getAttribute('data-fit') === '0') {
```

Comment:
As I understand this, looks like the host is going to change the wrapper attribute named `data-fit` so this triggers in the replayer. I think a better approach would be using the `CustomEvent` in the host and listening to it in the guests.

On the host you're gonna call the function `addCustomEvent` ([declared here](https://github.com/rrweb-io/rrweb/blob/924a4c898615d89c8c1cb240370c9e0ab7edaf73/typings/record/index.d.ts#L4)). Then, you can listen to custom events with `replayer.on(ReplayerEvents.CustomEvent, <callback>)` [like this](https://github.com/gitduckhq/gitduck/blob/3bf5d775f57cd67ac625a1c8a4574ed13d68b355/teletype-dom/lib/GuestDomBinding.ts#L46).

Reply:
Also, using `CustomEvent` there is no need to change the sources here